### PR TITLE
Add CONTRIBUTING.md file

### LINKED ISSUE #8: Metamath software future directions
Currently the development of metamath.exe is orphanized.  The discussion in pull request https://github.com/metamath/set.mm/pull/2392 showed, there is interest in moving forward, and there exist already some ideas under the hood.

This issue is meant to evaluate the current state of affairs, and bundle further discussions in a single location, instead of hijacking other threads for this purpose.

To me, there seems to be neither a team dedicated to, nor matured results about, future directions of Metamath.  Lets change this starting a discussion from scratch.  As far as I am concerned, I will be busy over Xmas, but will have spare time afterwards.

Please post your ideas, regardless of whether they are blurred or laser-sharp.  I am looking forward to learn about your visions.

Ideally, I should have started with my own ideas first, but I am in a bit of a hurry right now. Sorry.

Wolf

### LINKED ISSUE #13: A word of caution from Norm
Digging through my older emails, I found an email I received from Norm in 2017, from which I extract the following word of caution. It was before the move to GitHub, but some parts are still relevant.  This is not a real "Issue" in GitHub's sense, but I thought I'd leave it open here for a while.

> Another issue with the metamath program that people have requested is putting it on GitHub.  However, strange as it may seem I actually don't want to encourage major contributions at this point in time.  Almost every unsolicited patch I've been sent has had to be rewritten by me to fix memory leaks and other issues, and it can take a lot of my time to analyze them in complete detail.  C is a dangerous language if you don't have intimate knowledge of all details of the program's data structures etc.  If I put it on GitHub, I would dread having someone make a massive change that would require a week or more of my time to analyze.  Ensuring bug-free code can take more time than it took to write it.  I keep past versions archived and will put them on GitHub when that day comes.

### PR DESCRIPTION
**Pull request by jkingdon**
Most of the content here is adapted from various comments by various
people on github issues and pull requests. It isn't intended to be a
significant departure from how we have been working for the past month
or so.

Fixes #8 
Fixes #13 

One administrative note: @digama0 and @wlammen do we want to merge #41 first? If so, I'd be glad to do the merge between those changes and this one. Or if we think that one isn't ready, then this one could go first.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ongoing changes to our processes or metamath proof databases (especially
+those in the https://github.com/metamath/set.mm repository), we do not
+encourage developing significant new functionality in metamath-exe.
+
+Why? metamath.exe is the oldest of the metamath tools and is showing its
+age in a lot of ways including that C is probably not the language that
+anyone would choose for this kind of code today.
+
+Here's what Norm Megill, who wrote metamath-exe and did almost all of
+its maintenance until 2021, said:
```
Comment (digama0) [inline CONTRIBUTING.md:18]:
```suggestion
its maintenance until his passing in 2021, said:
```
It is helpful to give context on why the maintenance has changed so suddenly.

**Comment by wlammen** [comment]
Fine with me.  Don't be bothered about the merge conflict with #41.  I'll fix that.

**Review by digama0 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (jkingdon) [reply CONTRIBUTING.md]: I thought about wording like that and wasn't sure but OK, I'll accept this suggestion. We can always re-revise it later if it somehow the context feels different in a year or two.

**Comment by jkingdon** [comment]
Given two approvals I'll go ahead and merge this.

As usual, this doesn't have to be the end of review - if anyone sees anything they'd like to say differently, we can always open follow up pull requests.

**Comment by jkingdon** [comment]
> Given two approvals I'll go ahead and merge this.
> 
> As usual, this doesn't have to be the end of review - if anyone sees anything they'd like to say differently, we can always open follow up pull requests.

Oh wait I don't seem to have write access. I'm not sure whether I'll be doing enough metamath-exe work to quite need it, but feel free to either give me that access or merge this pull request yourself @digama0

**Pull request merged**